### PR TITLE
[USMP] Update RFC with constants pools

### PR DESCRIPTION
**Pull request by manupak**
This commit introduces the notion of constant memory pools
and removes the need to define access on each of the targets.

### CODE REVIEW

**Comment by manupa-arm** [comment]
cc : @areusch

File: rfcs/0009_Unified_Static_Memory_Planning.md [hunk 1]
```diff
@@ -461,19 +461,43 @@ Inputs :
...
+            Map<Target, Integer> target_burst_bytes;
+            /*! \brief The targets linked to the pool */
+            Array<Target> targets;
         }
+
+        struct WorkspacePoolInfoNode : public PoolInfoNode {
+        };
         ```
-        The input IRModule is expected to have "candidate_memory_pools" annotation populated with a orderered list of PoolInfo objects. The ordering will indicate to the planner the order of preference each allocate will be pinned to each Pool. The core compiler will run a pass to assign each tir.allocate with candidate_memory_pools based on the target each PrimFunc is executed, prior to invoking the USMP.
+        The input IRModule is expected to have "candidate_memory_pools" annotation populated with a orderered list of WorkspacePoolInfo objects. The ordering will indicate to the planner the order of preference each allocate will be pinned to each Pool. The core compiler will run a pass to assign each tir.allocate with candidate_memory_pools based on the target each PrimFunc is executed, prior to invoking the USMP.
```
Comment (areusch) [inline rfcs/0009_Unified_Static_Memory_Planning.md:500]: not sure if we should wade into the ordering/prioritization problem here. i mean we could, but it does seem like a bit of a can of worms. we could also solve this by customizing the allocator. just curious if you have a use case in mind?

**Review by areusch — APPROVED** [review]
thanks @manupa-arm , just a question here which i might have missed before

File: rfcs/0009_Unified_Static_Memory_Planning.md (hunk 1 above)
Comment (manupa-arm) [reply rfcs/0009_Unified_Static_Memory_Planning.md:500]: thanks @areusch for taking a look.

The ordering/prioritization is not in the scope of this change.

In fact, it is something already supported : https://github.com/apache/tvm/blob/395e91ff54543864a90240d18c8efd8c277c758b/tests/python/unittest/test_tir_usmp_algo.py#L366-L413

However, the USMP will only try to solve the allocation problem not the multi-objective problem of memory and performance. In other words, the scheduling stage is responsible for assigning the right pool for each allocate node. However, if the scheduling does not care about what pools it gets allocated, USMP will try to do a prioritized/ordered consumption of the pools to assign all allocated nodes to memory pools.

Comment (areusch) [reply rfcs/0009_Unified_Static_Memory_Planning.md:500]: ok, that makes sense to me

Comment (areusch) [reply rfcs/0009_Unified_Static_Memory_Planning.md:500]: only request--shall we document the above somewhere? maybe in a file-level docstring for USMP?

Comment (manupa-arm) [reply rfcs/0009_Unified_Static_Memory_Planning.md:500]: Sure, happy to do that in a follow up

**Comment by areusch** [comment]
i have no concerns here, @manupa-arm feel free to submit this when you're ready to start

**Pull request closed without merging**